be/src/storage/rowset/segment.cpp

```diff
@@ -123,6 +132,7 @@ Status Segment::parse_segment_footer(fs::ReadableBlock* rblock, SegmentFooterPB*
                     strings::Substitute("Bad segment file $0: failed to parse footer", rblock->path()));
         }
     } else { // Need read file again.
+        DCHECK(partial_rowset_footer == nullptr);
```

Comment:
This check is non-sense because you check it is a null pointer or not.

Reply:
done